test/functional/services/supertest.ts

```diff
@@ -29,12 +29,14 @@
  */
 
 import { FtrProviderContext } from 'test/functional/ftr_provider_context';
-import { format as formatUrl } from 'url';
 
 import supertestAsPromised from 'supertest-as-promised';
 
 export function OpenSearchDashboardsSupertestProvider({ getService }: FtrProviderContext) {
   const config = getService('config');
-  const opensearchDashboardsServerUrl = formatUrl(config.get('servers.opensearchDashboards'));
+  const opensearchDashboardsServerUrl = config
+    .get('servers.opensearchDashboards')
+    .toString()
+    .slice(0, -1);
```

Comment:
If `process.env.TEST_OPENSEARCH_DASHBOARDS_URL = 'http://miki.test:7777/osd'`, this slice with chop off the `d`.

Reply:
I will add a condition to check if the url is ends with "/". Will only do the modification if the url ends with "/".

Reply:
@AMoo-Miki  I think we don't need to add this ends with slash check here. We need to remove the pathname if `process.env.TEST_OPENSEARCH_DASHBOARDS_URL = 'http://miki.test:7777/osd` we shouldn't return a URL string with the pathname. Because we don't store the pathname part before. The url parts only include the protocol, hostname, port, auth, username and password. The url format result won't include the pathname.

Reply:
@AMoo-Miki My above comment was incorrect, URL.origin won't include the username and password part. I won't use this solution.